File: JumpGame2/Jumpgame2.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
    public: 
        int n;
        vector<int> arr;
        bool ans = true;

        Solution() {
            cin>>n;
            for(int i=0;i<n;i++) {
                int e; cin>>e;
                arr.emplace_back(e);
            }

            findSolution();
        }

        void findSolution();
        ~Solution() {
            if (ans) cout<<"true\n";
            else cout<<"false\n";
        }
};

void Solution::findSolution() {

    for (int i=0;i<n-1;i++) {
        if (arr[i] == 0) {
            bool flag = false;

            for (int j=0;j<i;++j) {
                if (arr[j]+j>i) {flag=true; break;}
            }

            if (!flag) {ans=false; return;}
        }
    }
}

int main() {
    Solution *obj = new Solution();
    delete obj;
}

/*
5
3 2 1 0 4
*/

File: JumpGame2/jumpGame2.py
class Solution(object):

    def __init__(self):
        self.n = int(input())
        self.arr = [int(num) for num in input().split()]
        self.ans = True

        self.findSolution()

    def __del__(self):
        if self.ans:
            print("Yes")
        else:
            print("No")

    def findSolution(self):
        for i in range(0, self.n - 1):
            if self.arr[i] == 0:
                flag = False
                for j in range(0, i):
                    if (self.arr[j]+j > i):
                        flag = True
                        break
                if flag == False:
                    self.ans = False
                    return


def main():
    sol = Solution()


if __name__ == "__main__":
    main()


'''
5
3 2 1 0 4
'''
